src/Refactoring-UI/ReExtractToTemporaryDriver.class.st

```diff
@@ -0,0 +1,145 @@
+Class {
+	#name : 'ReExtractToTemporaryDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'sourceInterval',
+		'newVariableName',
+		'selector',
+		'class',
+		'shouldEscape',
+		'variableNameEditorPresenterClass',
+		'stream'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'testing' }
+ReExtractToTemporaryDriver >> canEditName [
+	^ true
+]
+
+{ #category : 'testing' }
+ReExtractToTemporaryDriver >> canRenameArgs [
+	^ false
+]
+
+{ #category : 'execution' }
+ReExtractToTemporaryDriver >> changes [
+	
+	^ refactoring privateTransform; changes
+
+]
+
+{ #category : 'resources' }
+ReExtractToTemporaryDriver >> configureRefactoring [
+
+	refactoring := RBExtractToTemporaryRefactoring 
+							extract: sourceInterval
+						 	to: newVariableName
+							from: selector
+							in: class
+]
+
+{ #category : 'instance creation' }
+ReExtractToTemporaryDriver >> extract: anInterval to: aString from: aSelector in: aClass [
+
+	class := RBNamespace new classObjectFor: aClass.
+	selector := aSelector.
+	sourceInterval := anInterval.
+	newVariableName := aString
+]
+
+{ #category : 'initialization' }
+ReExtractToTemporaryDriver >> initialize [ 
+	
+	super initialize.
+	shouldEscape := false.
+]
+
+{ #category : 'accessing' }
+ReExtractToTemporaryDriver >> model: aModel [
+	model := aModel
+]
+
+{ #category : 'execution' }
+ReExtractToTemporaryDriver >> requestVariableName [
+
+	newVariableName := self defaultRequestDialog
```

Comment:
I think that source of your errors are the mocks. They are very tricky. Basically here we use `defaultRequestDialog`. We send it message `title:`, `label:` and `validateAnswer:`. We need mock for it that will answer for example `self` for first two, and on `validateAnswer:` it will answer `newVariableName` *that we want*.